Source/Dafny/DafnyOptions.cs

```diff
@@ -1056,6 +1060,9 @@ not sound
         /verificationLogger:trx;LogFileName=<...>.
     The exact mapping of verification concepts to the TRX and CSV formats is
     experimental and subject to change!
+/compileMocks 
+    If true, automatically compile external methods annotated with {{:mock}}
+    into the target language (currently only supported for C#)
```

Comment:
I have updated Attributes.md in the Reference Manual with a brief description of the `:synthesize` attribute (I renamed `:mock` to this).

My reasoning behind adding a command line option (now also renamed to `:synthesize`) is that synthesizing mocks is only possible with the Moq library, which we probably shouldn't require by default since it is not part of C#'s standard library. This command-line option would tell the compiler that it is OK to include Moq. 

Alternatively, one can traverse the program's AST and only try to import Moq if there is a method annotated with `:mock`. The problem with this approach is that most of the compiler code is agnostic to the program currently being compiled. This would mean that I would have to compile all fields as virtual (necessary for using Moq) even if no mocking is actually performed because I don't know which program I am compiling at the moment. @keyboardDrummer  previously mentioned that doing so might slow done the resuling binary, so I went for the command-line-argument approach. But I do have both approaches implemented locally, so I can easily make the change. Let me know if I should redo this one way or the other!